semantics/src/traversal/functions.rs

```diff
@@ -151,6 +166,28 @@ fn func_stmt(
     }
 }
 
+fn if_statement(
+    scope: Shared<FunctionScope>,
+    context: Shared<Context>,
+    stmt: &Spanned<fe::FuncStmt>,
+) -> Result<(), SemanticError> {
+    if let fe::FuncStmt::If {
+        test,
+        body: _,
+        or_else: _,
+    } = &stmt.node
+    {
+        let attributes = expressions::expr(scope, context, &test)?;
+        if let Type::Base(base) = attributes.typ {
```

Comment:
I would suggest:

```
if Type::Base(Base::Bool) == attributes.typ {
  return Ok(());
}
```

Reply:
Ah, of course. I sometimes still fail to recognize when there are nicer options available with Rust :)